src/Utilities/GenerateBatchVmSkus.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using Azure.Core;
+using Azure.Identity;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Batch;
+using Microsoft.Extensions.Configuration;
+using System.Reflection;
+using System.Text.Json;
+using Tes.Models;
+using Azure.ResourceManager.Compute;
+using Azure;
+using Azure.ResourceManager.Compute.Models;
+
+namespace TesUtils
+{
+    internal class Configuration
+    {
+        public string? SubscriptionId { get; set; }
+        public string? OutputFilePath { get; set; }
+
+        public static Configuration BuildConfiguration(string[] args)
+        {
+            var configBuilder = new ConfigurationBuilder();
+
+            var configurationSource = configBuilder.AddCommandLine(args).Build();
+            var configurationProperties = typeof(Configuration).GetTypeInfo().DeclaredProperties.Select(p => p.Name).ToList();
+
+            var invalidArguments = configurationSource.Providers
+                .SelectMany(p => p.GetChildKeys(Enumerable.Empty<string>(), null))
+                .Where(k => !configurationProperties.Contains(k, StringComparer.OrdinalIgnoreCase));
+
+            if (invalidArguments.Any())
+            {
+                throw new ArgumentException($"Invalid argument(s): {string.Join(", ", invalidArguments)}");
+            }
+
+            var configuration = new Configuration();
+            configurationSource.Bind(configuration);
+
+            return configuration;
+        }
+    }
+
+    internal class GenerateBatchVmSkus
+    {
+        static async Task Main(string[] args)
+        {
+            Configuration configuration = null;
+
+            try
+            {
+                configuration = Configuration.BuildConfiguration(args);
+            }
+            catch (ArgumentException ex)
+            {
+                Console.ForegroundColor = ConsoleColor.Red;
+                Console.WriteLine(ex.Message);
+                Console.ResetColor();
+                Environment.Exit(1);
+            }
+
+            Environment.Exit(await RunAsync(configuration));
+        }
+
+        static async Task<int> RunAsync(Configuration configuration)
+        {
+            var client = new ArmClient(new DefaultAzureCredential());
+            static double ConvertMiBToGiB(int value) => Math.Round(value / 1024.0, 2);
+            var subscription = client.GetSubscriptionResource(new ResourceIdentifier($"/subscriptions/{configuration.SubscriptionId}"));
+
+            var regionsForVm = new Dictionary<string, HashSet<string>>();
+            var sizeForVm = new Dictionary<string, VirtualMachineSize>();
+            var skuForVm = new Dictionary<string, ComputeResourceSku>();
+
+            foreach (var region in typeof(AzureLocation).GetTypeInfo().DeclaredProperties.Where(p => p.PropertyType == typeof(AzureLocation)).Select(p => p.Name).ToList())
```

Comment:
I like getting the regions from here but unfortunately it's missing some it seems.  For example, "westus3".  We need to find a better source for regions, or maybe update the NUGET?

Reply:
az account list-locations -o table

Reply:
you could check the az cli code to see which API it is using to get this list.  Notice it includes westus3

Reply:
Looks like this did the trick without having to run on a pay-as-you-go sub: 
await subscription.GetLocationsAsync().Where(x => x.Metadata.RegionType == RegionType.Physical).ToListAsync();